tests/test_authorization_code.py

```diff
@@ -81,17 +81,30 @@ def setUp(self):
     @httpretty.activate
     def test_happy_path(self):
         response = util.create_response()
-
         self.setup_expected_auth_code_token_request_response(200, response['wireResponse'])
 
         context = adal.AuthenticationContext(cp['authUrl'])
+
+        # action
         token_response = context.acquire_token_with_authorization_code(self.authorization_code, self.redirect_uri, response['resource'], cp['clientId'], cp['clientSecret'])
 
+        # assert
+
+        # the caching layer adds a few extra fields, let us pop them out for easier verification
+        for k in ['_clientId', '_authority', 'resource']:
+            token_response.pop(k)
```

Comment:
@yugangw-msft 
Thanks for identifying this little quirk and working it out!
Nitpicking: I would suggest to use a slightly safer way to rewrite this line to `token_response.pop(k, None)` because:
1. conceptually those tricky side effects are just implementation details, which could potentially be changed in future;
2. at the very least, this test case should not rely on an assumption that those side effects would always exist.
I will send out a new commit for it, if you don't mind.